src/main/java/entity/room/RoomFactory.java

```diff
@@ -0,0 +1,21 @@
+package entity.room;
+
+import java.io.IOException;
+import java.util.List;
+
+import entity.player.Player;
+
+/**
+ * Factory for creating game rooms.
+ */
+public interface RoomFactory {
```

Comment:
Again because CA has us creating so many specific Interactors, I'm not foreseeing much need for Factories. We always have a lot of context on what type of room we need when we will be instantiating it from any given Interactor (ie. it will be tied to a very specific use case).

Reply:
Again I made the room factory Generic. We can decide on whether to have a factory at all later.